improve readme and add contributing file (#122)
cloudflare/next-on-pages: docs/supported.md

| redirects                        | 🔄      |
| custom headers                   | 🔄      |
| custom page extensions           | 🔄 ?    |
| CDN support with asset prefix    | 🔄      |
| custom Image loader config       | 🔄      |
| custom webpack config            | 🔄 ?    |
| compression                      | 🔄 ?    |
| runtime configuration\*          | ❌      |
| disabling x-powered-by           | 🔄      |
| disabling ETag generation        | ✅ ?    |
| disabling HTTP keep-alive        | 🔄 ?    |
| setting custom build directory\* | ❌      |
| configuring the build id\*       | ❌      |
| configuring onDemandEntries      | ❌ ?    |
| ignoring ESLint                  | ✅ ?    |
| ignoring TypeScript errors       | ✅ ?    |
| exportPathMap                    | ❌      |
| trailing slash                   | 🔄      |
| react Strict Mode                | ✅ ?    |
| URL imports                      | ❌ ?    |
| build indicator\*                | ❌      |
| Turbopack-specific options\*     | ❌      |

    - ✅: Supported
    - 🔄: Not currently supported, but it's probably possible and we may add support in the future
    - ❌: Not supported and unlikely we ever will support this

>     - ?: Unsure support needs more investigation to confirm

Author: - [ ] TODO: Solve the `?`s before merging the PR

Author: I did check every option and I am pretty confident this list is correct 🙂

- **runtime configuration**: We could look into implementing the runtime configuration but it is probably not worth it since it is a legacy configuration and environment variables should be used instead.

- **setting custom build directory**: Applications built using next-on-pages don't rely on the `.next` directory so this option isn't really applicable.

- **configuring the build id**: Not applicable for next-on-pages applications.

> <!-- I assume that the build id is something vercel uses? -->

Author: - [ ] TODO: check this before merging the PR

## Build Output Configuration

> The following options have been gathered from the [Vercel Build Output API (v3) documentation](https://vercel.com/docs/build-output-api/v3#build-output-configuration):

| `config.json` property  | Support |
| ----------------------- | ------- |
| version                 | `3`     |
| routes `src`            | ✅      |
| routes `dest`           | 🔄      |
| routes `headers`        | 🔄      |
| routes `methods`        | ✅      |
| routes `continue`       | 🔄      |
| routes `caseSensitive`  | ✅      |
| routes `check`          | 🔄      |
| routes `status`         | 🔄      |
| routes `has`            | ✅      |
| routes `missing`        | ✅      |
| routes `locale`         | 🔄      |
| routes `middlewarePath` | ✅      |
| images<sup>1</sup>      | 🔄      |
| wildcard                | 🔄      |
| overrides               | 🔄      |
| cache                   | ❌      |

    - ✅: Supported
    - 🔄: Not currently supported, but it's probably possible and we may add support in the future
    - ❌: Not supported and unlikely we ever will support this

> - _1_ - **images**: If you want to use Next.js images you can provide your own [custom loader](https://nextjs.org/docs/api-reference/next/image#loader) which can also use Cloudflare images as indicated in the [Cloudflare's Image Integration documentation](https://developers.cloudflare.com/images/image-resizing/integration-with-frameworks/#nextjs).

Reviewer: ```suggestion
- _1_ - **images**: If you want to use `next/image`, you can provide your own [custom loader](https://nextjs.org/docs/api-reference/next/image#loader) and use Cloudflare Image Resizing, as per [Cloudflare's Image Resizing documentation](https://developers.cloudflare.com/images/image-resizing/integration-with-frameworks/#nextjs).
```

>   We hope to streamline this by adding a Cloudflare custom loader to next-on-pages at some point in the future.

Reviewer: ```suggestion
```

I wouldn't commit to this just yet.

Author: sure 🙂👍

| custom Image loader config                 | 🔄      |
| custom webpack config                      | ✅      |
| compression<sup>1</sup>                    | ❌      |
| runtime configuration<sup>2</sup>          | ❌      |
| disabling x-powered-by                     | 🔄      |
| disabling ETag generation                  | 🔄      |
| disabling HTTP keep-alive                  | ❌      |
| setting custom build directory<sup>3</sup> | ❌      |
| configuring the build id                   | ✅      |
| configuring onDemandEntries<sup>4</sup>    | ❌      |
| ignoring ESLint                            | ✅      |
| ignoring TypeScript errors                 | ✅      |
| exportPathMap<sup>5</sup>                  | ❌      |
| trailing slash                             | 🔄      |
| react Strict Mode<sup>6</sup>              | 🔄      |
| URL imports                                | ✅      |
| build indicator<sup>7</sup>                | ❌      |
| Turbopack-specific options<sup>8</sup>     | ❌      |

    - ✅: Supported
    - 🔄: Not currently supported, but it's probably possible and we may add support in the future
    - ❌: Not supported and unlikely we ever will support this

> - _1_ - **compression**: Locally no compression is applied, when deployed to Pages [Cloudflare applies gzip or brotli compression](https://developers.cloudflare.com/support/speed/optimization-file-size/what-will-cloudflare-compress) automatically.

Reviewer: ```suggestion
- _1_ - **compression**: [Cloudflare applies gzip or brotli compression](https://developers.cloudflare.com/support/speed/optimization-file-size/what-will-cloudflare-compress) automatically. When developing locally with Wrangler, no compression is applied.
```

- _2_ - **runtime configuration**: We could look into implementing the runtime configuration but it is probably not worth it since it is a legacy configuration and environment variables should be used instead.

- _3_ - **setting custom build directory**: Applications built using next-on-pages don't rely on the `.next` directory so this option isn't really applicable.

- _4_ - **configuring onDemandEntries**: Not applicable since its an option for the Next.js server during development which we don't rely on.

- _5_ - **React strict mode**: Currently we build the application so React strict mode doesn't work either way, if we can make strict mode work this option will most likely work straight away.

- _6_ - **exportPathMap**: Option used for SSG not applicable for next-on-pages apps.

> - _7_ - **build indicator**: If you're developing using `wrangler pages dev` we do hard refreshes so the indicator doesn't appear, if you run your app locally using `next dev` this option works fine.

Reviewer: ```suggestion
- _7_ - **build indicator**: If you're developing using `wrangler pages dev`, we do hard refreshes so the indicator doesn't appear. If you run your app locally using `next dev`, this option works fine.
```
